src/axolotl/utils/callbacks.py

```diff
@@ -514,3 +514,27 @@ def log_table_from_dataloader(name: str, table_dataloader):
             return control
 
     return LogPredictionCallback
+
+
+class SaveAxolotlConfigtoWandBCallback(TrainerCallback):
+    """Callback to save axolotl config to wandb"""
+
+    def __init__(self, cfg):
+        self.cfg = cfg
+
+    def on_train_begin(
+        self,
+        args: AxolotlTrainingArguments,  # pylint: disable=unused-argument
+        state: TrainerState,  # pylint: disable=unused-argument
+        control: TrainerControl,
+        **kwargs,  # pylint: disable=unused-argument
+    ):
+        if is_main_process():
+            try:
+                artifact = wandb.Artifact(name="axolotl-config", type="config")
+                artifact.add_file(local_path=self.cfg.axolotl_config_path)
+                wandb.run.log_artifact(artifact)
+                LOG.info("Axolotl config has been saved to WandB as an artifact.")
```

Comment:
maybe log.debug is enough

Reply:
Im indifferent, we log/print 20+ lines of Wandb initialization (and hundreds of lines on training begin) anyway, wouldn't it make sense to leave it at info to make people aware that their config is archived - how should anyone (except us) ever find/know it otherwise?
With an explicit (short) statement after successful archivation in the log, people then know that they can e.g safely overwrite their config.yml for a new experiment.

But happy to change that as well, whats your opinion @winglian ?